src/Token.php

```diff
@@ -20,11 +22,57 @@ class Token
 
     /**
      * @param string $value
-     * @param string $type
      */
-    public function __construct(string $value, string $type)
+    public function __construct(string $value)
     {
         $this->value = $value;
-        $this->type = $type;
+        $this->type = $this->getTokenType();
+    }
+
+    /**
+     * Returns StripeToken type base on incoming token.
+     *
+     * @return string
+     */
+    private function getTokenType(): string
```

Comment:
this belongs on the `PodPoint\Payments\Providers\Stripe\Token` class not this one